SpatialGDK/Source/Public/Utils/SchemaUtils.h

```diff
@@ -119,24 +119,39 @@ inline improbable::UnrealObjectRef Schema_IndexObjectRef(Schema_Object* Object,
 	ObjectRef.Offset = Schema_GetUint32(ObjectRefObject, 2);
 	if (Schema_GetBytesCount(ObjectRefObject, 3) > 0)
 	{
-		ObjectRef.Path = Schema_GetString(ObjectRefObject, 3);
+		ObjectRef.Path = GetStringFromSchema(ObjectRefObject, 3);
 	}
 	if (Schema_GetObjectCount(ObjectRefObject, 4) > 0)
 	{
-		ObjectRef.Outer = improbable::UnrealObjectRef(Schema_GetObjectRef(ObjectRefObject, 4));
+		ObjectRef.Outer = improbable::UnrealObjectRef(GetObjectRefFromSchema(ObjectRefObject, 4));
 	}
 
 	return ObjectRef;
 }
 
-inline improbable::UnrealObjectRef Schema_GetObjectRef(Schema_Object* Object, Schema_FieldId Id)
+inline improbable::UnrealObjectRef GetObjectRefFromSchema(Schema_Object* Object, Schema_FieldId Id)
 {
-	return Schema_IndexObjectRef(Object, Id, 0);
+	return IndexObjectRefFromSchema(Object, Id, 0);
 }
 
 using StringToEntityMap = TMap<FString, Worker_EntityId>;
 
-inline StringToEntityMap Schema_GetStringToEntityMap(Schema_Object* Object, Schema_FieldId Id)
+inline void AddStringToEntityMapToSchema(Schema_Object* Object, Schema_FieldId Id, StringToEntityMap& Map)
+{
+	if (Map.Num() == 0)
+	{
+		return;
+	}
+
+	for (auto& Pair : Map)
+	{
+		Schema_Object* PairObject = Schema_AddObject(Object, 1);
+		AddStringToSchema(PairObject, SCHEMA_MAP_KEY_FIELD_ID, Pair.Key);
+		Schema_AddEntityId(PairObject, SCHEMA_MAP_VALUE_FIELD_ID, Pair.Value);
```

Comment:
Shouldn't this function be called `AddEntityIdToSchema`?

Reply:
Nope, it's a part of the C API.